ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -420,40 +440,94 @@ - (void)testShouldHandleFileOpEvent {
+
+    org_vfsforgit_PrjFSProviderUserClient userClient;
+    VirtualizationRootHandle testRootHandle = InsertVirtualizationRoot_Locked(
+        &userClient,
+        0,
+        repoRootVnode.get(),
+        repoRootVnode->GetVid(),
+        FsidInode{ repoRootVnode->GetMountPoint()->GetFsid(), repoRootVnode->GetInode() },
+        repoPath.c_str());
+    XCTAssertTrue(VirtualizationRoot_IsValidRootHandle(testRootHandle));
 
+    int pid;
     // Fail when the provider is not online
     s_virtualizationRoots[0].providerUserClient = nullptr;
     XCTAssertFalse(
         ShouldHandleFileOpEvent(
             &perfTracer,
             context,
-            repoRootVnode.get(),
+            testVnodeFile.get(),
             nullptr, // path
             KAUTH_FILEOP_RENAME,
             true, // isDirectory,
             &testRootHandle,
             &pid));
-    s_virtualizationRoots[0].providerUserClient = &userClient;
+}
+
+- (void)testShouldHandleFileOpEvent_ProviderInitiatedIO {
+    PerfTracer perfTracer;
+    shared_ptr<mount> testMount = mount::Create();
+    std::string repoPath = "/Users/test/code/Repo";
+    shared_ptr<vnode> repoRootVnode = testMount->CreateVnodeTree(repoPath, VDIR);
+    shared_ptr<vnode> testVnodeFile = testMount->CreateVnodeTree(repoPath + "/file.txt");
+
+    org_vfsforgit_PrjFSProviderUserClient userClient;
+    VirtualizationRootHandle testRootHandle = InsertVirtualizationRoot_Locked(
+        &userClient,
+        0,
+        repoRootVnode.get(),
+        repoRootVnode->GetVid(),
+        FsidInode{ repoRootVnode->GetMountPoint()->GetFsid(), repoRootVnode->GetInode() },
+        repoPath.c_str());
+    XCTAssertTrue(VirtualizationRoot_IsValidRootHandle(testRootHandle));
 
     // Fail when pid matches provider pid
     MockProcess_Reset();
     MockProcess_AddContext(context, 0 /*pid*/);
     MockProcess_SetSelfPid(0);
     MockProcess_AddProcess(0 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+    int pid;
     XCTAssertFalse(
         ShouldHandleFileOpEvent(
             &perfTracer,
             context,
-            repoRootVnode.get(),
+            testVnodeFile.get(),
             nullptr, // path
             KAUTH_FILEOP_RENAME,
             true, // isDirectory,
             &testRootHandle,
             &pid));
-    MockProcess_Reset();
-    MockProcess_AddContext(context, 501 /*pid*/);
-    MockProcess_SetSelfPid(501);
-    MockProcess_AddProcess(501 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+}
+
+- (void)testShouldHandleFileOpEvent_VnodeCacheUpdated {
+    PerfTracer perfTracer;
+    shared_ptr<mount> testMount = mount::Create();
+    std::string repoPath = "/Users/test/code/Repo";
+    shared_ptr<vnode> repoRootVnode = testMount->CreateVnodeTree(repoPath, VDIR);
+    shared_ptr<vnode> testVnodeFile = testMount->CreateVnodeTree(repoPath + "/file.txt");
+    shared_ptr<vnode> testVnodeDirectory = testMount->CreateVnodeTree(repoPath + "/directory", VDIR);
```

Comment:
A lot of this code seems to be repeated in each of the newly-split tests. I guess that's why it was one giant test before.

I'm wondering if it'd be better to place these similar tests into an entirely new class where the boilerplate objects can be placed in ivars, and initialised/shut down in `- (void)setUp`/`- (void)tearDown`.